src/main/java/org/opensearch/flowframework/workflow/AbstractRetryableWorkflowStep.java

```diff
@@ -110,25 +91,36 @@ void retryableGetMlTask(
                 try {
                     logger.info(workflowStep + " successful for {} and modelId {}", workflowId, response.getModelId());
                     String resourceName = WorkflowResources.getResourceByWorkflowStep(getName());
+                    String id;
```

Comment:
This logic should ideally be in the Workflow Step classes themselves, rather than this abstract class, since we're dealing with outputs specific to those Workflow Step classes. If I recall correctly, this abstract class is just here to provide the setting which limits the maximum amount of times we can retry a request

Reply:
I'll remove this and only use modelId in follow up PR, do you mean then its okay to keep the status update call here in that case or move it regardless?

Reply:
I agree with @joshpalis that ideally an abstract class shouldn't have any specifics to a class.  Good for later refactoring.